django/db/models/functions/datetime.py

```diff
@@ -355,9 +355,7 @@ def convert_value(self, value, expression, connection):
                     "zone definitions for your database installed?"
                 )
         elif isinstance(value, datetime):
-            if value is None:
-                pass
-            elif isinstance(self.output_field, DateField):
+            if isinstance(self.output_field, DateField):
                 value = value.date()
             elif isinstance(self.output_field, TimeField):
                 value = value.time()
```

Comment:
It looks like [this whole branch](https://djangoci.com/job/django-coverage/HTML_20Coverage_20Report/z_183c4e2447e913dd_datetime_py.html#t357) is uncovered.

Reply:
Thanks didn't know we had this coverage job as well.

So it's not run on SQLite because `(datetime|date|time)_trunc_sql` run specialized Python functions that already return the proper time.

The first branch is implicitly tested by these usage of the `__date` transform in aggregation tests

https://github.com/django/django/blob/825ddda26a14847c30522f4d1112fb506123420d/tests/aggregation/tests.py#L2216

But we don't appear to have the equivalent coverage for `__time`